packages/analytics/src/rudderstack.ts

```diff
@@ -0,0 +1,153 @@
+import * as rudderanalytics from 'rudder-sdk-js';
+
+type SignupProvider = 'email' | 'phone' | 'google' | 'facebook' | 'apple';
+
+type VirtualSignupFormAction = {
+    action:
+        | 'open'
+        | 'started'
+        | 'email_confirmed'
+        | 'signup_continued'
+        | 'country_selection_screen_opened'
+        | 'password_screen_opened'
+        | 'signup_done'
+        | 'signup_flow_error'
+        | 'go_to_login';
+    signup_provider?: SignupProvider;
+    form_source?: string;
+    form_name?: string;
+    error_message?: string;
+};
+
+type RealAccountSignupFormAction = {
+    action:
+        | 'open'
+        | 'step_passed'
+        | 'save'
+        | 'restore'
+        | 'close'
+        | 'real_signup_error'
+        | 'other_error'
+        | 'real_signup_finished';
+    step_codename?: string;
+    step_num?: number;
+    user_choice?: string;
+    source?: string;
+    form_name?: string;
+    real_signup_error_message?: string;
+    landing_company: string;
+};
+
+type VirtualSignupEmailConfirmationAction = {
+    action: 'received' | 'expired' | 'confirmed' | 'error';
+    signup_provider?: SignupProvider;
+    form_source?: string;
+    email_md5?: string;
+    error_message?: string;
+};
+
+type TradeTypesFormAction =
+    | {
+          action: 'open' | 'close' | 'info_close';
+          trade_type_name?: string;
+          tab_name?: string;
+          form_source?: string;
+          form_name?: string;
+          subform_name?: string;
+      }
+    | {
+          action: 'choose_trade_type';
+          subform_name: 'info_old' | 'info_new';
+          form_name: string;
+          trade_type_name: string;
+      }
+    | {
+          action: 'choose_trade_type';
+          subform_name: 'trade_type';
+          tab_name: string;
+          form_name: string;
+          trade_type_name: string;
+      }
+    | {
+          action: 'search';
+          search_string: string;
+      }
+    | {
+          action: 'info_open';
+          tab_name: string;
+          trade_type_name: string;
+      }
+    | {
+          action: 'info-switcher';
+          info_switcher_mode: string;
+          trade_type_name: string;
+      };
+
+type IdentifyAction = {
+    language: string;
+};
+
+type TEvents = {
+    ce_virtual_signup_form: VirtualSignupFormAction;
+    ce_real_account_signup_form: RealAccountSignupFormAction;
+    ce_virtual_signup_email_confirmation: VirtualSignupEmailConfirmationAction;
+    ce_trade_types_form: TradeTypesFormAction;
+    identify: IdentifyAction;
+};
+
+export class RudderStack {
+    has_identified = false;
+    has_initialized = false;
+    current_page = '';
```

Comment:
@adrienne-deriv This is the old implementation, But do we really need these? 🤔 Have you tried to call `track` for example before `load` to see what happens? Would it crash or is the library doing error handling? We can probably remove these 🤔 

Same for `current_page`, What's the reason for keeping the last value? If the `pageView` is called shouldn't we just send the event? 🤔

Reply:
wait lemme try this out

Reply:
There's no error appearing in the console when we `track` without calling `load`, but the Rudderstack documentation general setup and recommendation is to call `load` first before calling other operations. This is similar to why we also maintain the `has_identified` variable, is to ensure that when we call `track`, we call it before `identifyEvent` is called, otherwise rudderstack will assign an anonymous ID to that tracked event instead of the user's ID.

Reply:
@adrienne-deriv So we don't need these variables then, Right?

We always call `load` and then `identify` and then `track`, Right now we are not doing any error handling or instantiating if `has_initialized` or `has_identified` are false, We just ignore the function call, Which as you checked it won't throw any error. 

What do you think?

Reply:
Noo load is called once when the app is mounted, `has_initialized` is set to true after the callback from `ready` is called, and these are done once only and we need the `has_initialized` to ensure that the SDK has been properly loaded with the rudderstack token and urls, only then we can call `page` and `track` and `identify`

Reply:
`has_identified` is set to true and is necessary to ensure we don't send events (call `page` and `track`)  without the user's information (that is initialised when the user logs in and `authorize` is called and `identify` event is called)

Reply:
Otherwise rudderstack will assign an anonymous ID if we don't call `identify` event yet or ensure that that event has been called after we load rudderstack by `load`
<img width="867" alt="Screenshot 2023-06-20 at 5 34 46 PM" src="https://github.com/binary-com/deriv-app/assets/103016120/986c83e5-6295-435d-9af6-e26fdaa3ee9d">

Reply:
But let me clarify with denis first if they allow us to track users that has not logged in yet, or they just want us to track users that has logged in

Reply:
Okies from Denis we need to only track users that has logged in, so `is_authorized` is required here to ensure that we only track users that has logged in